docs/theming-flatpak.rst

```diff
@@ -0,0 +1,56 @@
+Theming Flatpak
+===============
+
+Flatpak apps default to Adwaita as a theme, as they cannot use the system theme. This happens because all flatpaks are sandboxed and do not have the ability to read data files or libraries in ``/usr`` (where system themes are typically located). This is done so that the format is portable across a wide range of distributions in a reproducible environment, but has the downside of not being able to pick up themes. The solution to this was to package GTK themes as Flatpaks, as relying upon the host to have the correct version for every flatpak defeats the portability benefits it provides. These themes are provided as extensions to the Gnome runtime.
+
+Installing themes
```

Comment:
What you mean here is that if I had e.g. Greybird installed already on my computer, Flatpak would eventually install the Greybird theme all by itself? I guess if I had it installed but not currently being used though, and were using another Gtk theme instead, I'd still have the problem where the Flatpak apps would show with Adwaita - is that right?
Also which are the DConf values described below? I don't believe they are in my commited file, and they don't seem to be in your later comments either.

Reply:
> What you mean here is that if I had e.g. Greybird installed already on my computer, Flatpak would eventually install the Greybird theme all by itself? I guess if I had it installed but not currently being used though, and were using another Gtk theme instead, I'd still have the problem where the Flatpak apps would show with Adwaita - is that right?

Correct.

> Also which are the DConf values described below? 

Just open `dconf-editor` and navigate to the path, everything should be clear.

Reply:
One last question, but how I can find the path (and by extension these values) in dconf-editor? I see the various categories apps, ca, com e.t.c, and tried to search for the term 'path', but only found the paths of various applications. Also I knew that the path (when generally used) is this folder under home where commandline apps can be put and will be detected by the shell (in my case home/lmintmate/bin), but I doubt you mean that path. Just asking because I want to see these Dconf values, as they might be important enough to be included in the documentation.

Reply:
> I see the various categories apps, ca, com 

Those aren't categories those are path segments.

Reply:
So, are these the Dconf values at which Flatpak looks to determine which themes are installed in order to install the Flatpak equivalents at update time? For the record, for the moment I've written at that point of the text (amongst other phrases) "As of Flatpak 0.10.1, the Flatpak system can detect whether the system themes available correspond to any Flatpak themes available in the repositories, and, if so, will automatically install found themes at update time based upon various DConf values." I guess I could say "the various Dconf values" if it looks at all of them, but I doubt it works like that, as that would be an overkill. Should I leave it vague like that, or specify at which specific Dconf values Flatpak looks?
P.S. I've finished implementing all your other suggestions otherwise, and will push everything after seeing what I'll do with this one, as this was an extensive rewrite and would show all previous text segments as obsolete.

Reply:
It is probably useful to say the exact key I suppose (it is `gtk-theme` if not obvious). It also only ever includes a single theme, the current one.

Reply:
Since the gtk-theme key only ever includes the currently being used theme, how does Flatpak find the rest of the themes installed that have Flatpak versions available? E.g. if Greybird is currently used, would Flatpak spot that e.g. Ambiance is also installed via the same Dconf key? Or does the detecting of the rest of the themes happens on the basis of whether they have ever been used, which means that if Ambiance is installed but hasn't been enabled as a theme, the Flatpak system might not install its Flatpak version (until it gets enabled that is)?

Reply:
Related question, now that it occurs to me (somewhat belatedly perhaps, but still): I see now that the second to last paragraph of the text, the one starting with "GT3 on X11" (from your blog) is content-wise related to the Applying Themes subsection. I wonder something though: In the current Flatpak version, does Gtk3 on X11 also picks up the theme from the Dconf key, or does it still use XSettings? This matters because the next Ubuntu LTS (and Linux Mint as a consequence) won't transition to Wayland (because Wayland isn't quite there yet in terms of stability), which means that very many users will continue to use X11 for at least another two years (or even more if they continue to use the LTS until the end of its support in both distros). In general I also wonder just how outdated this second-to-last paragraph is in comparison to the current state of Flatpak, just so that the oudated parts are removed.

Reply:
It doesn't, that is what I said.

> does Gtk3 on X11 also picks up the theme from the Dconf key, or does it still use XSettings? 

AFAICT it only uses XSettings (we could perhaps change this upstream...). The GNOME XSettings daemon `gsd-xsettings` does read the DConf values and convert them into the XSettings values though.